cmd/apiserver/main.go

```diff
@@ -88,3 +99,18 @@ func main() {
 	log.Printf("Listening on: 127.0.0.1:%v", config.Port)
 	log.Fatal(srv.ListenAndServe())
 }
+
+func apiKeyFunc() httplimit.KeyFunc {
+	ipKeyFunc := httplimit.IPKeyFunc("X-Forwarded-For")
+
+	return func(r *http.Request) (string, error) {
+		v := r.Header.Get("X-API-Key")
+		if v != "" {
+			dig := sha1.Sum([]byte(v))
```

Comment:
This is probably overkill since it's all in-memory, but once we switch to Redis, we'll want to apply something like this in case someone gains unauthorized access to Redis.